goosebit/api/v1/software/routes.py

```diff
@@ -69,11 +71,16 @@ async def post_update(_: Request, file: UploadFile | None = File(None), url: str
         # local file
         artifacts_dir = Path(config.artifacts_dir)
         file_path = artifacts_dir.joinpath(file.filename)
-
-        async with aiofiles.tempfile.NamedTemporaryFile("w+b") as f:
-            await f.write(await file.read())
-            absolute = await file_path.absolute()
-            software = await create_software_update(absolute.as_uri(), Path(f.name))
+        tmp_file_path = artifacts_dir.joinpath("tmp", ("".join(random.choices(string.ascii_lowercase, k=12)) + ".tmp"))
```

Comment:
This change makes the artifact directory mandatory. Before, it was optional (when only using external artifacts) as temporary files were created in `/tmp` :disappointed:

Reply:
Didn't even think about this.  Maybe I can generate a temporary file location other than this?

Reply:
IMHO it is best to use a standard temp location. But, we should make sure it works on all OSes.